lib/mock/mock-call-history.js

```diff
@@ -0,0 +1,254 @@
+'use strict'
+
+const {
+  kMockCallHistoryAddLog,
+  kMockCallHistoryGetByName,
+  kMockCallHistoryClearAll,
+  kMockCallHistoryDeleteAll
+} = require('./mock-symbols')
+const { InvalidArgumentError } = require('../core/errors')
+
+const computingError = 'error occurred when computing MockCallHistoryLog.url'
+
+function makeFilterCalls (parameterName) {
+  return (parameterValue) => {
+    if (typeof parameterValue !== 'string' && !(parameterValue instanceof RegExp) && parameterValue != null) {
+      throw new InvalidArgumentError(`${parameterName} parameter should be one of string, regexp, undefined or null`)
+    }
+    if (typeof parameterValue === 'string' || parameterValue == null) {
+      return this.logs.filter((log) => {
+        return log[parameterName] === parameterValue
+      })
+    }
+    if (parameterValue instanceof RegExp) {
+      return this.logs.filter((log) => {
+        return parameterValue.test(log[parameterName])
+      })
+    }
+
+    return []
+  }
+}
+function computeUrlWithMaybeSearchParameters (requestInit) {
+  // path can contains query url parameters
+  // or query can contains query url parameters
+  try {
+    const url = new URL(requestInit.path, requestInit.origin)
+
+    // requestInit.path contains query url parameters
+    // requestInit.query is then undefined
+    if (url.search.length !== 0) {
+      return url
+    }
+
+    // requestInit.query can be populated here
+    url.search = new URLSearchParams(requestInit.query).toString()
+
+    return url
+  } catch {
+    // should never happens
+    return computingError
```

Comment:
Let's better throw here with an error that has as `cause` whatever is being catched at the `catch` scope

Reply:
done